ocp_utilities/infra.py

```diff
@@ -354,3 +357,56 @@ def create_icsp(icsp_name, repository_digest_mirrors):
     )
     icsp.deploy()
     return icsp
+
+
+def create_update_pull_secret(new_content_dict, ps_name, ps_namespace):
+    """Updating existing pull-secrets / Creating new pull-secrets
+
+    Args:
+        new_content_dict (dict): Secret data to be added/created,
+            e.g.: {"auths":{
+                    <registry_name>:
+                        {"auth": <auth_token>,
+                        "email": <auth_email>},
+                    ...,
+                    <registry_name>:
+                        {"auth": <auth_token>,
+                        "email": <auth_email>}
+                    }
+                }
+        ps_name (str): Pull-secret name
+        ps_namespace (str): Pull-secret namespace
+
+    Returns:
+        pull_secret: A new pull-secret with new_content_dict as secret data if
+            the pull-secret does not exist, else an updated pull-secret with
+            new_content_dict data addition is returned.
+    """
+    pull_secret = Secret(name=ps_name, namespace=ps_namespace)
+
+    if pull_secret.exists:
+        ps_data_dict = json.loads(
+            base64.b64decode(pull_secret.instance.data[".dockerconfigjson"])
+        )["auths"]
+        ps_data_dict.update(new_content_dict["auths"])
+        ps_data_dict_encoded = base64.b64encode(
+            json.dumps({"auths": ps_data_dict}).encode("ascii")
+        ).decode("utf-8")
```

Comment:
duplicate code; please extract to a function.

Reply:
done